internal/nostr/nostr.go

```diff
@@ -203,6 +203,59 @@ func (svc *Service) InfoHandler(c echo.Context) error {
 	}
 }
 
+func (svc *Service) PublishHandler(c echo.Context) error {
+	var requestData PublishRequest
+	if err := c.Bind(&requestData); err != nil {
+		return c.JSON(http.StatusBadRequest, ErrorResponse{
+			Message: fmt.Sprintf("error decoding publish request: %s", err.Error()),
+		})
+	}
+
+	if (requestData.SignedEvent == nil) {
+		return c.JSON(http.StatusBadRequest, ErrorResponse{
+			Message: "signed event is empty",
+		})
+	}
+
+	ctx, cancel := context.WithTimeout(c.Request().Context(), 90*time.Second)
+	defer cancel()
+
+	relay, isCustomRelay, err := svc.getRelayConnection(ctx, requestData.RelayUrl)
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, ErrorResponse{
+			Message: "error connecting to relay: " + err.Error(),
+		})
+	}
+
+	if isCustomRelay {
+		defer relay.Close()
+	}
+
+	svc.Logger.WithFields(logrus.Fields{
+		"eventId":   requestData.SignedEvent.ID,
+		"relayUrl":  requestData.RelayUrl,
+	}).Info("Publishing event...")
+
+	err = relay.Publish(ctx, *requestData.SignedEvent)
+	if err != nil {
+		svc.Logger.WithError(err).WithFields(logrus.Fields{
+			"eventId":   requestData.SignedEvent.ID,
+			"relayUrl":  requestData.RelayUrl,
+		}).Error("Failed to publish event")
+	
+		return c.JSON(http.StatusInternalServerError, ErrorResponse{
+			Message: "error publishing the event: " + err.Error(),
+		})
```

Comment:
can we provide some more details here? 
what are good status codes and what is a useful response JSON for the API consumer/developer.

Reply:
Modified ErrorResponse type to always have error, so users can check for resp.error and confirm if the result is successful

```golang
type ErrorResponse struct {
	Message string `json:"message"`
	Error   string `json:"error"`
}
```

Only using `StatusBadRequest 400`, `StatusRequestTimeout 408` and `StatusInternalServerError 500`

Reply:
great. 
maybe also compare it to the other APIs we have - can we take some inspiration from there (though we are not really good there, too)